lib/p2p/Pool.ts

```diff
@@ -191,11 +195,26 @@ class Pool extends EventEmitter {
 
     this.loadingNodesPromise = this.nodes.load();
     this.loadingNodesPromise.then(async () => {
-      if (this.nodes.count > 0 && !this.disconnecting) {
+      if (this.disconnecting) {
+        this.loadingNodesPromise = undefined;
+        return;
+      }
+
+      const { primary, secondary } = this.nodes.rank();
+      if (secondary.length > 0) {
+        this.secondaryPeersTimeout = setTimeout(async () => {
```

Comment:
I think comments would be good here to explain what's going on so we don't forget going forward (and also to ensure I understand right now). How about something like

`We delay connection attempts to secondary peers - whom we've had trouble connecting to in the past - to prevent overwhelming xud at startup`